src/hooks/V2V3TerminalVersion.ts

```diff
@@ -0,0 +1,52 @@
+import { readNetwork } from 'constants/networks'
+import { CV2V3 } from 'models/v2v3/cv'
+import { useEffect, useState } from 'react'
+import { ContractJson } from 'utils/v2v3/loadV2V3Contract'
+
+export function useV2V3TerminalVersion(
+  address: string | undefined,
+): CV2V3 | undefined {
+  const [v2Address, setV2Address] = useState<string>()
+  const [v3Address, setV3Address] = useState<string>()
+
+  useEffect(() => {
+    async function loadV2() {
+      const address = (
+        (await import(
+          `@jbx-protocol/contracts-v2-latest/deployments/${readNetwork.name}/JBETHPaymentTerminal.json`
+        )) as ContractJson
+      ).address
+
+      try {
+        setV2Address(address)
+      } catch (e) {
+        console.error('Failed to load V2 JBETHPaymentTerminal', e)
+      }
+    }
```

Comment:
**suggestion** to save some code, can probably get away with using the existing `loadJuiceboxV2Contract`, `loadJuiceboxV3Contract`, or `useLoadV2V3Contract` and specify the CV ([example](https://github.com/jbx-protocol/juice-interface/blob/76b135157034cca6936c413bebefa4a434e19c60/src/utils/safe.ts#L48-L55))